index.js

```diff
@@ -115,27 +115,38 @@ class TimeoutError extends Error {
 	}
 }
 
-const delay = ms => new Promise((resolve, reject) => {
+const safeTimeout = (resolve, reject, ms) => {
```

Comment:
Detaching promises is not a good idea. I'd use `try`/`catch` here, but then this piece:

```js
try {
    safeTimeout(resolve, ms);
} catch (error) {
    reject(error);
}
```

would repeat twice. So to follow the DRY principle, I'm good with the current solution :+1: